### ISSUE TITLE
mkbundle on windows faild , need help

### ISSUE THREAD

**Issue by azraelrabbit** [mono/mono#6769]
## Steps to Reproduce

1. i make a very simple app just to output "hello word". and then mkbundle it on windows.
2.  by fixed some compiling error ,i get the bundled exe file ,and can run in  ASCII path correct.
3. then i put the bundled exe to an chinese named  folder, does not work.
![mkbundle-faild](https://user-images.githubusercontent.com/553991/35757836-ddb17fb2-08ab-11e8-9a39-27d31a3c970f.png)

 
## Current Behavior

 as it say: Cannot determine the text encoding for the assembly location: E:\测试2\test.exe
Please add the correct encoding to MONO_EXTERNAL_ENCODINGS and try again.

i changed  MONO_EXTERNAL_ENCODINGS to any i known encoding , like utf-8
but not work.

and 

## Expected Behavior

 wanted it works fine on windows and non-ASCII encoding .

and another choise i change the temp.c file  to add this line to  main(int argc, char* argv[] )

SetEnvironmentVariable("MONO_EXTERNAL_ENCODINGS","UTF16");

but the application was broken by this message: 
..\mono\metadata\object.c:4677: String conversion error: Illegal byte sequence encounted in the input. assembly:<unknown assembly> type:<unknown type> member:<none>

the pre-version of 5.4.0 that set the variable MONO_EXTERNAL_ENCODINGS works fine. 
but the new version can not working any more. 

any one could help me?

### On which platforms did you notice this

[ ] macOS
[ ] Linux
[ X] Windows

**Version Used**:
windows 10  1709 x64

E:\测试2>mono -V
Mono JIT compiler version 5.8.0 (Visual Studio built mono)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           normal
        SIGSEGV:       normal
        Notification:  Thread + polling
        Architecture:  x86
        Disabled:      none
        Misc:          softdebug
        LLVM:          supported, not enabled.
        GC:            sgen

### Stacktrace

```
Please paste the stacktrace here if available.
```

**Comment by migueldeicaza** [comment]
Did you use the `--i18n` flag to specify all the encodings that you would use?

Try adding `--i18n all` first

**Comment by azraelrabbit** [comment]
@migueldeicaza 
thanks for your comment.
after i add the --i18n all  args.
there has another warning: 

E:\测试2>tt2.exe
The assembly mscorlib.dll was not found or could not be loaded.
It should have been installed in the `E:\lib\mono\4.5\mscorlib.dll' directory.

after i copy the mscrolib.dll to  the `E:\lib\mono\4.5\mscorlib.dll' directory,

that warning again:

E:\测试2>tt2.exe

Cannot determine the text encoding for the assembly location: E:\测试2\test.exe
Please add the correct encoding to MONO_EXTERNAL_ENCODINGS and try again.

**Comment by migueldeicaza** [comment]
If you are getting `mscorlib.dll` not found, your bundle is incorrect.

**Comment by HinTak** [comment]
Yes, your bundle is wrong. Please show the full output of `mkbundle -v ...` with the `-v` switch. That should tell you where mscorlib is copied from.
